client/sttp-client/src/main/scala/sttp/tapir/client/sttp/EndpointToSttpClient.scala

```diff
@@ -96,6 +99,20 @@ private[sttp] class EndpointToSttpClient[R](clientOptions: SttpClientOptions, ws
     l.flatMap(leftParams => r.map(rightParams => combine(leftParams, rightParams)))
   }
 
+  private def outputMatchesContent(output: EndpointOutput[_], content: Either[String, MediaType]): Boolean = {
+    def mediaMatchesContent(media: MediaType): Boolean =
+      content match {
+        case Right(mt) => media.noCharset == mt.noCharset
+        case Left(_)   => false
+      }
+
+    output match {
```

Comment:
here again, the output can be mapped, e.g. `jsonBody.map(identity)` wouldn't work. So using `.traverseOutputs` or looking at the basic inputs would solve the problem.